core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -56,7 +60,9 @@ class LogManager(val logDirs: Array[File],
                  scheduler: Scheduler,
                  val brokerState: BrokerState,
                  brokerTopicStats: BrokerTopicStats,
-                 time: Time) extends Logging {
+                 val zkUtils: ZkUtils,
```

Comment:
@junrao I realized that there is one way to do this without using `LogManger.setLogFailureCallback ()`. We can provide `KafkaServer` to the constructor of `LogManager`. Then `LogManager.handleLogDirFailure()` can call `kafkaServer.replicaManager.handleLogDirFailure()`. Does this sound reasonable?